php-checks/src/main/java/org/sonar/php/checks/HttpOnlyCheck.java

```diff
@@ -56,7 +56,25 @@
   @Override
   public void visitFunctionCall(FunctionCallTree tree) {
     if (isSetCookie(tree) && httpOnlySetToFalse(tree)) {
-      context().newIssue(this, tree.callee(), MESSAGE).secondary(tree.arguments().get(HTTP_ONLY_PARAMETER_INDEX), null);
+    
+      if (tree.arguments().size() > HTTP_ONLY_PARAMETER_INDEX) {
+        context().newIssue(this, tree.callee(), MESSAGE).secondary(tree.arguments().get(HTTP_ONLY_PARAMETER_INDEX), null);
+      }
+      else if(tree.arguments().size() != 3) {
+        // if argument is not defined
+        
+        // but if there is exactly 3 arguments
+        // maybe this call was used:
+        // $params["expire"] = "xx";
+        // $params["httponly"] = true;
+        // ect
+        // setcookie($name, $value, $params); // compliant
+        
+        // or this one
+        // setcookie($name, $value, $expire); // non compliant
```

Comment:
As for the other rule, I would remove this long comment and make sure that the tests are explicit about the FPs and FNs.